### PR TITLE
Student group creation

### PR DESCRIPTION
**Pull request by Zlash65**
Summary:
1. Targeting issue #9251 , updated creation_tool code to fetch and populate each group with the list of students that fall in that group.
2. Improvised tests to fetch instructor name(code) from db.
3. Wrong test data being stored - rectified (test_student_group.js)
4. Removed duplicate code. Changed frappe.throw from student_group.py to frappe.msgprint.

### CODE REVIEW

File: erpnext/schools/doctype/student_group_creation_tool/student_group_creation_tool.py [hunk 1]
```diff
@@ -67,6 +67,49 @@ def create_student_groups(self):
...
 			student_group.academic_year = self.academic_year
+			student_list = get_students(self.academic_year, d.group_based_on, self.academic_term, self.program, d.batch, d.course)
+			if student_list:
+				for student in student_list:
+					student_group.append('students', student)
 			student_group.save()
 
 		frappe.msgprint(_("{0} Student Groups created.".format(l)))
+
+@frappe.whitelist()
```
Comment (manassolanki) [inline erpnext/schools/doctype/student_group_creation_tool/student_group_creation_tool.py]: @Zlash65 Please don't write the duplicated code. You can import the function from the student group.

File: erpnext/schools/doctype/student_group/student_group.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def get_students(academic_year, group_based_on, academic_term=None, program=None
 			student_list.append(s)
 		return student_list
 	else:
-		frappe.throw(_("No students found"))
+		frappe.msgprint("No students found")
```
Comment (manassolanki) [inline erpnext/schools/doctype/student_group/student_group.py]: @Zlash65 Use translatable message like `frappe.msgprint(_("Message"))`

File: erpnext/schools/doctype/student_group_creation_tool/student_group_creation_tool.py [hunk 3]
```diff
@@ -67,6 +68,11 @@ def create_student_groups(self):
 			student_group.max_strength = d.max_strength
 			student_group.academic_term = self.academic_term
 			student_group.academic_year = self.academic_year
+			student_list = get_students(self.academic_year, d.group_based_on, self.academic_term, self.program, d.batch, d.course)
+			print student_list
```
Comment (manassolanki) [inline erpnext/schools/doctype/student_group_creation_tool/student_group_creation_tool.py]: Remove print statement

File: erpnext/schools/doctype/student_group_creation_tool/student_group_creation_tool.py [hunk 4]
```diff
@@ -67,6 +68,11 @@ def create_student_groups(self):
 			student_group.max_strength = d.max_strength
 			student_group.academic_term = self.academic_term
 			student_group.academic_year = self.academic_year
+			student_list = get_students(self.academic_year, d.group_based_on, self.academic_term, self.program, d.batch, d.course)
+
+			if student_list:
+				for student in student_list:
```
Comment (rmehta) [inline erpnext/schools/doctype/student_group_creation_tool/student_group_creation_tool.py]: `get_students` should return empty list if no students found.

File: erpnext/schools/doctype/student_group/test_student_group.js [hunk 5]
```diff
@@ -10,45 +10,54 @@ QUnit.test('Test: Student Group', function(assert){
...
-					]);
-				});
-			});
-			return frappe.run_serially(tasks);
+			return frappe.tests.make('Student Group', [
+				{academic_year: '2016-17'},
+				{academic_term: '2016-17 (Semester 1)'},
+				{program: "Standard Test"},
+				{group_based_on: 'Batch'},
+				{student_group_name: loop[0]},
```
Comment (rmehta) [inline erpnext/schools/doctype/student_group/test_student_group.js]: change variable name

**Review by rmehta — CHANGES_REQUESTED** [review]

**Pull request merged**